ArduCopter/mode_auto.cpp

```diff
@@ -1970,15 +1970,17 @@ void ModeAuto::do_RTL(void)
 /********************************************************************************/
 
 // verify_takeoff - check if we have completed the takeoff
-bool ModeAuto::verify_takeoff()
+bool ModeAuto::verify_takeoff(const AP_Mission::Mission_Command& cmd)
 {
-#if AP_LANDINGGEAR_ENABLED
     // if we have reached our destination
     if (auto_takeoff_complete) {
+#if AP_LANDINGGEAR_ENABLED
         // retract the landing gear
         copter.landinggear.retract_after_takeoff();
-    }
 #endif
+        // notify user
+        sendtext_command_complete(cmd.index);
```

Comment:
This bothers me a bit because the verify is having a side-effect.  Currently we don't have a situation where verify will be called repeatedly if it returns true.   I think.  It's also not a new problem as we're doing it in existing places.

But perhaps we should look at the caller of these verify methods announcing the thing being reached (so removing the announcements  from `verify_nav_wp` where you've currently changed to using your new method.

Reply:
I've talked with the Partner again and pointed them at the MISSION_CURRENT message and my impression is that they will move to use that instead so we may not need to merge this... although it does no harm really.

Reply:
OK, I've clarified the real reason for this PR.  There's apparently no way to know what the current active command is in Auto mode when looking at the **Dataflash logs**.

Reply:
> OK, I've clarified the real reason for this PR. There's apparently no way to know what the current active command is in Auto mode when looking at the **Dataflash logs**.

Yes and no.

https://github.com/ardupilot/ardupilot/blob/master/ArduCopter/mode_auto.cpp#L604

1.  It's in the wrong spot (Copter not AP_Mission)
2. it's the same message we use to log the changes to the stored mission....